samplesheets/tests/test_templatetags.py

```diff
@@ -0,0 +1,225 @@
+            comments=DEFAULT_COMMENTS,
+        )
+        # Set up Assay
+        self.assay = self.make_assay(
+            file_name=ASSAY_FILE_NAME,
+            study=self.study,
+            tech_platform=ASSAY_TECH_PLATFORM,
+            tech_type=ASSAY_TECH_TYPE,
+            measurement_type=ASSAY_MEASURE_TYPE,
+            arcs=[],
+            comments=DEFAULT_COMMENTS,
+        )
+        # Setup iRODS backend for the test
+        self.irods_backend = get_backend_api('omics_irods')
+
+    def test_get_investigation(self):
+        """Test get_investigation()"""
+        self.assertEqual(
+            s_tags.get_investigation(self.project), self.investigation
+        )
+
+    def test_get_investigation_no_investigation(self):
+        """Test get_investigation() without investigation"""
+        self.investigation.delete()
+        self.assertEqual(s_tags.get_investigation(self.project), None)
+
+    def test_get_search_item_type_with_material_types(self):
+        """Test get_search_item_type() with material types"""
+        for material_type in GENERIC_MATERIAL_TYPES:
+            item = {'type': material_type}
+            expected_type = GENERIC_MATERIAL_TYPES[material_type]
+            self.assertEqual(s_tags.get_search_item_type(item), expected_type)
+
+    def test_get_search_item_type_with_file(self):
+        """Test get_search_item_type() with special case 'file'"""
+        item = {'type': 'file'}
+        self.assertEqual(s_tags.get_search_item_type(item), 'Data File')
+
+    def test_get_irods_tree(self):
+        """Test get_irods_tree()"""
+        ret = s_tags.get_irods_tree(self.investigation)
+        # Assert that IRODS_SAMPLE_COLL exists in the returned string
+        self.assertIn(IRODS_SAMPLE_COLL, ret)
+        # Assert that study path exists in the returned string
+        self.assertIn(self.irods_backend.get_sub_path(self.study), ret)
+        # Assert that assay path exists in the returned string
+        clean_assay_path_0 = self.irods_backend.get_sub_path(self.assay).split(
+            '/'
+        )[0]
+        clean_assay_path_1 = self.irods_backend.get_sub_path(self.assay).split(
+            '/'
+        )[1]
+        self.assertIn(clean_assay_path_0, ret)
+        self.assertIn(clean_assay_path_1, ret)
+
+    def test_get_material_search_url(self):
+        """Test get_material_search_url()"""
+        item = {'study': self.study, 'name': 'Sample1'}
+        url = s_tags.get_material_search_url(item)
+        expected_url = reverse(
+            'samplesheets:project_sheets',
+            kwargs={'project': self.project.sodar_uuid},
+        )
+        expected_url += '#/study/{}/filter/Sample1'.format(
+            self.study.sodar_uuid
+        )
+        self.assertEqual(url, expected_url)
+
+    def test_get_irods_path_with_project(self):
+        """Test get_irods_path() with project"""
+        expected_path = self.irods_backend.get_path(self.project)
```

Comment:
We can just name these variables `expected`, unless there are multiple expected values within a single test cases we need to differentiate between.